### PR TITLE
happy: Update

### PR DESCRIPTION
**Pull request by essandess**
happy: Update

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [  ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/happy/Portfile [hunk 1]
```diff
@@ -4,27 +4,27 @@ PortSystem          1.0
...
 homepage            https://github.com/simonmar/happy
 
 description         A parser generator for Haskell
-long_description    Happy is a parser generator for Haskell. Given a\
-                    grammar specification in BNF, Happy generates\
-                    Haskell code to parse the grammar. Happy works in\
+long_description    Happy is a parser generator for Haskell. Given a \
+                    grammar specification in BNF, Happy generates \
+                    Haskell code to parse the grammar. Happy works in \
                     a similar way to the yacc tool for C.
```
Comment (reneeotten) [inline devel/happy/Portfile:19]: please don't make unneeded white-space changes as they clutter the diff to see the actual functional changes

File: devel/happy/Portfile [hunk 2]
```diff
@@ -4,27 +4,27 @@ PortSystem          1.0
...
 
 master_sites        https://hackage.haskell.org/package/${name}-${version}
 
 conflicts           hs-happy
```
Comment (reneeotten) [inline devel/happy/Portfile:23]: please check what `port lint` has to say here

Comment (essandess) [reply devel/happy/Portfile]: Thanks deleted.

File: devel/happy/Portfile (hunk 1 above)
Comment (essandess) [reply devel/happy/Portfile]: Removed.

**Comment by reneeotten** [comment]
I noticed that this port (and a few others in the PR queue) are downloading stuff (i.e., Preparing to download ghc-8.10.4 ...). I am not familiar with this type of ports, but is that really necessary and can you not just a dependency on `ghc`? I am probably not really qualified to review/merge these PRs though due to lack of knowledge about it...

**Comment by essandess** [comment]
`stack` downloads its own build tools and Haskell libraries. That’s how it avoids the problem of version hell and provides robust builds.

> I noticed that this port (and a few others in the PR queue) are downloading stuff (i.e., Preparing to download ghc-8.10.4 ...). I am not familiar with this type of ports, but is that really necessary and can you not just a dependency on `ghc`? I am probably not really qualified to review/merge these PRs though due to lack of knowledge about it...

**Pull request merged**